File: apps/connect4/src/app/features/jobs/models/Stats.ts
export class Stats {
  constructor(
    public gameTime: number,
    public doesRedWin: boolean | null,
    public redDeepness: number,
    public yellowDeepness: number,
    public moves: Move[]
  ) {
  }
}

export class Move {
  constructor(
    public coordinate: Coordinate,
    public isRedTurn: Boolean,
    public time: number
  ) {
  }
}

export class Coordinate {
  constructor(
    public x: number,
    public y: number
  ) {
  }
}

File: apps/connect4-bot/src/main/kotlin/Stats.kt
package org.example

data class Stats(
    var gameTime: Long = 0,
    var doesRedWin: Boolean? = null,
    var redDeepness: Int = 1,
    var yellowDeepness: Int = 1,
    var moves: MutableList<Move> = mutableListOf()
)

data class Move(
    var time: Long = 0,
    var isRedTurn: Boolean = true,
    var coordinate: Coordinate = Coordinate(),
)

data class Coordinate(
  var x: Int = 0,
  var y: Int = 0
)
